### ISSUE TITLE
NDViewer crash on Windows when clicking buttons with PySide6>=6.7

### ISSUE THREAD

**Issue by jacopoabramo** [pyapp-kit/ndv#59]
* ndv version: main branch (locally cloned)
* Python version: 3.10
* Operating System: Windows 11
* Installation options: pip install .[pyside,vispy]

### Description

I'm currently launching the viewer with a local script after installing ndv from my fork. Whenever I click the "composite" or "3D/2D" buttons, the script just crashes with no thrown exception or error message.

Crash after clicking composite...

https://github.com/user-attachments/assets/031a2d52-fb32-4f2b-b789-19915e80a5df

... and after clicking 3D/2D.

https://github.com/user-attachments/assets/9a3b46a7-9429-4b1c-bb9b-2c604ee7d287

### What I Did

test script on Windows:

```python
import ndv
from qtpy.QtWidgets import QApplication
from ndv import NDViewer

images = ndv.data.cells3d()

app = QApplication([])
viewer = NDViewer(images)
viewer.show()
app.exec_()
```

Is it a bug I forgot to notice in the backlog?

**Comment by jacopoabramo** [comment]
Forgot to add: when running in debug mode and putting breakpoints in the incriminated code sections (a.k.a. the callbacks of those pushbuttons), it reaches the breakpoint with no problem and keeps going.

**Comment by tlambert03** [comment]
As mentioned on zulip, please try pyqt6 and pyside6<6.8 and see if you still get it

**Comment by tlambert03** [comment]
This may also be an issue with OpenGL on your system... to debug, it would be helpful to override sys.excepthook so that it hopefully doesn't crash out to console (but the fact that no trace back shows suggests this is an even lower level crash).  Will try to reproduce locally when I can

**Comment by tlambert03** [comment]
yeah, can confirm same behavior here, pyside6 not working well on windows.  (pyside6 has generally always been harder to support :/)  i'll leave this open until i have time to debug.  i almost always use pyqt6, so I need to also check on macos linux.  we do test against them all on ci. but that doesn't always catch the same stuff
